tests_and_analysis/test/script_tests/utils.py

```diff
@@ -46,15 +46,18 @@ def get_fig_label_data(fig) -> Dict[str, Union[str, List[str]]]:
     label_data = {'x_ticklabels': [],
                   'title': fig._suptitle.get_text()}
 
-    # Get axis labels from whichever axis has them; collect all tick labels
+    # Get axis labels from whichever axis has them.
+    # Collect all tick labels from visible axes only,
+    # we don't care about invisible axis labels
     for ax in fig.axes:
-        ax_label_data = get_ax_label_data(ax)
+        if ax.get_frame_on():
+            ax_label_data = get_ax_label_data(ax)
```

Comment:
Would it make sense to add something like:
```python
else:
    # Invisible axes should not have visible ticks or labels
    assert not any([tickline.get_visible() for tickline in ax.get_xticklines()])
    assert not any([tickline.get_visible() for tickline in ax.get_yticklines()])
    assert all(np.array([t.get_color() for t in ax.get_xticklabels()]) == 'none')
    assert all(np.array([t.get_color() for t in ax.get_yticklabels()]) == 'none')
```
?

Reply:
I guess there's no harm in it!

Reply:
I had to change it a bit to avoid line breaks but done :)

Reply:
Oh no, seems like `ax.get_x/yticklines().get_visible()` returns `True` for old versions of `matplotlib`, even though you can't see them... maybe we can skip this check?

Reply:
Argh! Ok, a bit more experimentation suggests a cleaner overall strategy. If in the plotter we have:

```python
for ax_dir in ax.xaxis, ax.yaxis:
    ax_dir.set_major_locator(plt.NullLocator())
```

then there shouldn't _be_ any ticks at all on the invisible axis. Then instead of looking at properties, we can just

```python
for ticklines in ax.get_xticklines(), ax.get_yticklines():
    assert len(ticklines) == 0
```

and be a _lot_ more confident that they won't find some crazy way to gain a label and/or become visible.

Reply:
Ok, so if we use `NullLocator` for the y axis (or set `yticks=[]` in `add_subplot`), we get this weird result where the y label overlaps the ticks: 
![image](https://user-images.githubusercontent.com/18187336/196201322-cc03e633-5159-4fa1-801d-efaed6f9ae94.png)

I think that's because the `ylabel` is attached to the invisible axis, so if there are no tick labels on the invisible axis it puts the label close to the axis as if the visible ticks weren't there. I think we have 2 options:

* Set x-axis locator only to `NullLocator`, don't check y-axis ticks
* Set x and y-axis locators to `NullLocator`, set y-label on one of the visible axes instead (this should be fine as they both have `sharey=True`

Reply:
I've tried the second one in [9259f49](https://github.com/pace-neutrons/Euphonic/pull/243/commits/9259f49f9af6e4bc35354ea6adc782774d188758), fingers crossed...

Reply:
Wait, of course this would mess up the xlabels too, argggggh!

![image](https://user-images.githubusercontent.com/18187336/196208409-d4b5838e-8dad-470c-9b1c-d1181922ba8c.png)

Reply:
> I suppose, then, that our best option is to
> 
> * not check for invisibility
> * wait for a user report and fix the problem when it occurs. Perhaps by then we can drop some MPL versions and check more consistently.

I think so, it's been a good lesson of don't touch plotting unless you really have to...